### PR TITLE
idris2: new port in lang

### PR DESCRIPTION
**Pull request by barracuda156**
#### Description

New port in lang

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.6
Xcode 3.2

macOS 14.5
Xcode 15.4

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? 
- [ ] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

File: lang/idris2/Portfile [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+github.tarball_from archive
+
+depends_build-append \
+                    port:chez-scheme \
+                    path:libexec/coreutils/libstdbuf.so:coreutils
+
+depends_lib-append  port:gmp
+
+# Used at runtime too:
+depends_run-append  port:chez-scheme
```
Comment (reneeotten) [inline lang/idris2/Portfile:27]: being a build- and runtime-dependency is the definition of a "lib" dependency. So just add it once there instead.

**Review by reneeotten — CHANGES_REQUESTED** [review]

File: lang/idris2/Portfile (hunk 1 above)
Comment (barracuda156) [reply lang/idris2/Portfile:27]: I do not think it is correct, strictly: lib dependency implies linking. Though I am aware that it is often used that way due to lazy coding :)

Comment (barracuda156) [reply lang/idris2/Portfile:27]: @reneeotten (No big deal, of course, if you really want lib dep syntax here, I will change that.)

Comment (barracuda156) [reply lang/idris2/Portfile]: @reneeotten Anyway, I changed that to lib, rather save time.

File: lang/idris2/Portfile [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+PortSystem          1.0
+PortGroup           github 1.0
+PortGroup           makefile 1.0
+
+name                idris2
+github.setup        idris-lang Idris2 0.7.0 v
+categories          lang
+license             BSD
+maintainers         {@barracuda156 gmail.com:vital.had} openmaintainer
+description         Purely functional programming language with first class types
```
Comment (pmetzger) [inline lang/idris2/Portfile:12]: Just FYI, this is a terrible description. What makes Idris special is that it allows real programming with dependent types, so you can prove your programs are correct in the type system.

**Comment by pmetzger** [comment]
Just wanted to say it will be cool to have an Idris port.

File: lang/idris2/Portfile (hunk 2 above)
Comment (barracuda156) [reply lang/idris2/Portfile:12]: I just copied it, of course, not authored. I will change it according to your suggestion.

Comment (barracuda156) [reply lang/idris2/Portfile]: Done.

**Comment by pmetzger** [comment]
@reneeotten Are you happy at this point?

**Pull request merged**